samples/Sentry.Samples.Console.Basic/Program.cs

```diff
@@ -33,66 +36,133 @@
 
     // This option tells Sentry to capture 100% of traces. You still need to start transactions and spans.
     options.TracesSampleRate = 1.0;
-});
 
-// This starts a new transaction and attaches it to the scope.
-var transaction = SentrySdk.StartTransaction("Program Main", "function");
-SentrySdk.ConfigureScope(scope => scope.Transaction = transaction);
+#if NET6_0_OR_GREATER
+    // This option tells Sentry to capture a heap dump when the process uses more than 5% of the total memory. The heap
+    // dump will be sent to Sentry as a file attachment.
+    options.EnableHeapDumps(5);
+
+    // This determines the level of heap dump events that are sent to Sentry
+    options.HeapDumpEventLevel = SentryLevel.Warning;
 
-// Do some work. (This is where you'd have your own application logic.)
-await FirstFunction();
-await SecondFunction();
-await ThirdFunction();
+    // A debouncer can be configured to tell Sentry how frequently to send heap dumps. In this case we've configured it
+    // to capture a maximum of 3 events per day and to wait at least 1 hour between each event.
+    options.HeapDumpDebouncer = Debouncer.PerDay(3, TimeSpan.FromHours(1));
```

Comment:
I played around with different APIs. I wanted to minimise the confusion between the ThreasholdTriggerPercentage amd the TriggerFunction... to make it really clear you either provided one or the other. So rather than exposing an object that has both of those properties, there are now two overloads of SentryOptions.EnableHeapdumps that basically wrap that options class with what is relevant for each variant.